pytests/tests/test_path.py

```diff
@@ -16,3 +16,15 @@ def test_take_pathbuf():
 def test_take_pathlib():
     p = pathlib.Path("/root")
     assert rpath.take_pathbuf(p) == str(p)
+
+
+def test_take_pathlike():
```

Comment:
I think it would also be good to test the case where the argument is not path-like and where it is path-like but `__fspath__` does not return `str` or `bytes`.

Reply:
Done.